### PR TITLE
chore: Update templates provided to aid in collaboration and following the process

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?
- Update templates provided (contributing, issue, and PR) to bring more awareness to process and expectations for contributing/collaborating
- [x] ✋ I am aware of the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests

### Motivation
- Encourage details and discussions in issues first before setting out on implementation. Do we need some "mechanism" to say an issue is fleshed out and ready for implementation or just a maintainer giving a "sounds good, feel free to submit a PR if interested", ?

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -1,23 +1,30 @@
 ---
 name: Feature request
 about: Suggest an idea for this project
-title: '[FEATURE] <title>'
```
Comment (bryantbiggs) [inline .github/ISSUE_TEMPLATE/feature_request.md:4]: we use labels so no need for the templated `[FEATURE]`

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
-
 ---
 
-#### Is your feature request related to a problem? Please describe
-A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
+<!--- Please keep this note for the community --->
+
+### Community Note
+
+* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
```
Comment (bryantbiggs) [inline .github/ISSUE_TEMPLATE/feature_request.md:10]: pinched from Terraform AWS provider - just a more explicit statement of what we want users to do in terms of crowdsourcing feature request priority

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- Module version [Required]:
+
+- Terraform version:
+<!-- Execute terraform -version -->
+- Provider version(s):
+<!-- Execute: terraform providers -version -->
+
+## Reproduction Code [Required]
+
+<!-- REQUIRED -->
```
Comment (bryantbiggs) [inline .github/ISSUE_TEMPLATE/bug_report.md:33]: hard requirement, HARD requirement

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+Steps to reproduce the behavior:
```
Comment (bryantbiggs) [inline .github/ISSUE_TEMPLATE/bug_report.md:35]: very important as well - these two pieces give us the closes representation of a reproduction

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,26 +1,25 @@
-
 ### What does this PR do?
 
-<!-- A brief description of the change being made with this pull request. -->
+- [ ] ✋ I am aware of the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests
```
Comment (bryantbiggs) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: Lets see if this helps reinforce the practice to flesh out details through issues first before committing to a large body of code in a PR

**Review by askulkarni2 — APPROVED** [review]
Thanks for putting this together @bryantbiggs!

**Pull request merged**